game-core/src/main/java/games/strategy/triplea/delegate/battle/StrategicBombingRaidBattle.java

```diff
@@ -663,18 +658,13 @@ private void removeAaHits(
       final IDelegateBridge bridge, final CasualtyDetails casualties, final String currentTypeAa) {
     final List<Unit> killed = casualties.getKilled();
     if (!killed.isEmpty()) {
-      bridge
-          .getHistoryWriter()
-          .addChildToEvent(
-              MyFormatter.unitsToTextNoOwner(killed) + " killed by " + currentTypeAa,
-              new ArrayList<>(killed));
       final IntegerMap<UnitType> costs = TuvUtils.getCostsForTuv(attacker, gameData);
       final int tuvLostAttacker = TuvUtils.getTuv(killed, attacker, costs, gameData);
       attackerLostTuv += tuvLostAttacker;
       // attackingUnits.removeAll(casualties);
       removeAttackers(killed, false);
-      final Change remove = ChangeFactory.removeUnits(battleSite, killed);
-      bridge.addChange(remove);
+      new KillUnits(battleSite, killed, "${units} killed by " + currentTypeAa + " in ${territory}")
```

Comment:
The formatting of the message pattern being a responsibility of the caller is concerning to me. This creates a coupling between the data and the formatting strings used internally by 'killUnits' and the caller. The caller must be aware of the string type. It does seem like the formatted string could become an implementation detail of `KillUnits` by adding a third constructor, eg:
```new KillUnits(battleSite, killed, currentTypeAa)```

At that point too I'd recommend to add a `@Builder` on that constructor so that the meaning of the parameters is more clear when reading the calling context. 

WDYT?

Reply:
`KillUnits` has a constructor that is already `Territory, Collection<Unit>, String` so I can't add this third constructor.  I could remove the existing one but there might be another caller that will have a different string to be used and then I'll be back to the same problem.

I like the idea of hiding the string as an implementation detail.  I guess I could create a child class of `KillUnits` that is `KillAaCasualtyUnits` (or something like that) and then it can have this special constructor.  Are you ok with that?

Reply:
Multiple constructors with the same signature usually calls for a factory pattern. In that case just create static methods that create the new object and they can be different in their naming. Would that work?